src/spi/mod.rs

```diff
@@ -1,8 +1,42 @@
 //! SPI traits
 
+use crate::private::Sealed;
+
 pub mod blocking;
 pub mod nb;
 
+/// Marker trait for SPI Word
+pub trait SpiWord: Sealed {
+    /// Specified data size type
+    type Data: Sized;
+}
+
+/// 8-bit SPI Word (default)
+pub type U8 = u8;
+/// 9-bit SPI Word
+pub struct U9;
+/// 16-bit SPI Word
+pub struct U16;
+/// 18-bit SPI Word
+pub struct U18;
+
+impl Sealed for U9 {}
+impl Sealed for U16 {}
```

Comment:
i think i would prefer not to alias `u8` and `u16` so that it's still obvious from the interface these are primitive types.
the other question i have is whether these should be _markers_ or _actual data types_, and whether approach this can adequately describe non-byte word sizes.

it seems to me that if you're running in 9-bit mode you're more than likely to be using arrays of 16-bit aligned words, and with 18-bit mode either 24-bit (if you have sub-word addressing) or 32-bit aligned words (if you do not). On one hand, it makes sense to be able to run over say `&[U9]`, on the other, you may need to control the representation of this in memory for things like DMA.

i wonder whether an approach similar to I2C might describe this better, though i also don't think this covers all cases and it _definitely_ makes the bounds more complex...

```rust
trait SpiWord {
  type Data;
}

impl SpiWord for u8 {
  type Data = u8;
}

impl SpiWord for U9 {
  type Data = u16;
}
```